### DISCUSSION TITLE
Is a useStaticQuery Hook Monitored for Incremental Builds?

### DISCUSSION THREAD

**Discussion by badaczewski** [gatsbyjs/gatsby#36207] [Help]
It appears to me that useStaticQuery **hooks** are not monitored by GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES. I have a hook that gets reused within components to load Drupal menu items. Even though the data source for the menu items has changed Gatsby build using the previous query results. I'm I correct that these are not monitored? Will making this a usStaticQuery on a page template solve this?

P.S - Anyone know how to cache bust just the hook?

Called within component like:
```typescript
import { useMainMenu } from "../hooks/main-menu"
```

The contents of the hook itself at /src/hooks/main-menu.tsx
```typescript
import { useStaticQuery, graphql } from "gatsby"

export const useMainMenu = () => {
  const { allMenuItems } = useStaticQuery(
    graphql`
      query MainMenu {
        allMenuItems(filter: {enabled: {eq: true}}) {
          edges {
            node {
              title
              url
              id
              parent {
                id
              }
            }
          }
          totalCount
        }
      }
    `
  )
  return allMenuItems.edges
}
```

**Comment by laneparton** [comment]
I've had similar experiences and am interested in the best practice here.
